### PR TITLE
Grab rework

### PR DESCRIPTION
**Pull request by vode-code**
## About The Pull Request
Reworked grab delay to use updating the sprite instead of a sprite animation so that the delay is clear to players even with varying ROB values.
Added delay between passive and aggressive grab, along with between kill grab starting and ending.
Deleted that twenty-ish second stun neck grab had.

## Why It's Good For The Game
For the rework, the animations were heavily separate from what was actually going on and that was bad.
For the removal of the twenty second stun, people could stunlock with that if they felt like it and that is also bad.
Also... Baycode is Badcode.

## Changelog
:cl: Chickenish
balance: Grabbing now takes time to upgrade aggressive and kill grabs.
balance: Neck Grab no longer stuns the victim for twenty seconds.
fix: Grab animations now line up with their effects.
/:cl:

### CODE REVIEW

**Comment by vode-code** [comment]
WHY IS THE SPAM NOT GONE?

**Comment by SirRichardFrancis** [comment]
https://git-scm.com/book/en/v2

File: code/modules/mob/mob_grab.dm [hunk 1]
```diff
@@ -229,50 +230,72 @@
 		if(EAST)
 			animate(affecting, pixel_x =-shift, pixel_y = 0, 5, 1, LINEAR_EASING)
 
+/obj/item/grab/proc/upgrade_grab(delay_time, hud_icon_state_after, state_after)
+	if(!allow_upgrade)
+		return // upgrading now is not allowed!
+	delay_time = round(delay_time/8) // the sprites have eight configurations for timer
```
Comment (SirRichardFrancis) [inline code/modules/mob/mob_grab.dm:236]:
```suggestion
	delay_time = round(delay_time / 8) // the sprites have eight configurations for timer
```

File: code/modules/mob/mob_grab.dm [hunk 2]
```diff
@@ -229,50 +230,72 @@
...
+	var/original_icon = hud.icon_state
+	var/original_loc = get_turf(assailant) // used to see if the assailant moved and thus disrupted the upgrade
+	var/original_time = world.time
+	for(var/counter in 1 to 8)
+		sleep(delay_time)
+		if((last_action > original_time) || (!confirm()) || (get_turf(assailant) != original_loc)) // cannot do a grab attack while upgrading a grab
```
Comment (SirRichardFrancis) [inline code/modules/mob/mob_grab.dm:242]:
```suggestion
		if(last_action > original_time || !confirm() || get_turf(assailant) != original_loc) // cannot do a grab attack while upgrading a grab
```

File: code/modules/mob/mob_grab.dm [hunk 3]
```diff
@@ -229,50 +230,72 @@
...
-			msg_admin_attack("[assailant] pinned down [affecting].")
-			apply_pinning(affecting, assailant)
-
-		state = GRAB_AGGRESSIVE
+		assailant.visible_message(SPAN_WARNING("[assailant] is securing \his grip on [affecting]!"))
 		icon_state = "grabbed1"
-		hud.icon_state = "reinforce1"
+		hud.icon_state = "reinforce"
+		state = GRAB_SECURING
+		if(upgrade_grab(total_warmup,"reinforce_final", GRAB_AGGRESSIVE))
```
Comment (SirRichardFrancis) [inline code/modules/mob/mob_grab.dm:283]:
```suggestion
		if(upgrade_grab(total_warmup, "reinforce_final", GRAB_AGGRESSIVE))
```

File: code/modules/mob/mob_grab.dm [hunk 4]
```diff
@@ -229,50 +230,72 @@
...
-	// Adjust the grab cooldown using assailant's ROB stat
+	// Adjust the grab warmup using assailant's ROB stat
 	var/assailant_stat = assailant.stats.getStat(STAT_ROB)
-	var/cooldown_increase
+	var/warmup_increase
 	if(assailant_stat > 0)
-		// Positive ROB decreases cooldown, but not linearely
-		cooldown_increase = -(assailant_stat ** 0.8)
+		// Positive ROB decreases warmup, but not linearly
+		warmup_increase = -(assailant_stat ** 0.8)
```
Comment (SirRichardFrancis) [inline code/modules/mob/mob_grab.dm:269]:
```suggestion
		warmup_increase -= assailant_stat ** 0.8
```

File: code/modules/mob/mob_grab.dm [hunk 5]
```diff
@@ -229,50 +230,72 @@
...
 	if(!assailant.canmove || assailant.lying)
 		qdel(src)
 		return
 
-	// Adjust the grab cooldown using assailant's ROB stat
+	// Adjust the grab warmup using assailant's ROB stat
 	var/assailant_stat = assailant.stats.getStat(STAT_ROB)
-	var/cooldown_increase
+	var/warmup_increase
 	if(assailant_stat > 0)
```
Comment (SirRichardFrancis) [inline code/modules/mob/mob_grab.dm:267]:
```suggestion
	if(assailant_stat)
```

**Review by SirRichardFrancis — APPROVED** [review]

File: code/__DEFINES/mobs.dm [hunk 6]
```diff
@@ -19,6 +19,7 @@
 
 // Grab levels.
 #define GRAB_PASSIVE    1
+#define GRAB_SECURING 1.5
```
Comment (TheShown911) [inline code/__DEFINES/mobs.dm:22]: bruh

**Review by hyperioo — APPROVED** [review]
Let's see how it goes

**Pull request merged**